main.c

```diff
@@ -116,6 +116,17 @@ static void pointer_handle_enter(void *data, struct wl_pointer *wl_pointer,
 	if (output == NULL) {
 		return;
 	}
+
+	// the places the cursor moved away from are also dirty
+	// TODO: This should dirty outputs the current value
```

Comment:
isn't slurp single-threaded? is it even possible for `render()` to be called at that point?

Reply:
Oh - good question. I had just assumed it's multithreaded since it's handling events and setting callbacks, but I have no idea how Wayland works really. I could just take these TODOs out and things will be fine in practice.

Reply:
@emersion would know better than me, but my understanding is that it is single threaded, but asynchronous. So the render would happen on a subsequent iteration of the loop that calls `wl_display_dispatch`.